shim/src/interpose.c

```diff
@@ -58,7 +58,7 @@
                                                               \
         bool reentrant = is_reentrant_demi_call();            \
                                                               \
-        if ((!initialized) || (reentrant))                    \
+        if (reentrant)                                        \
```

Comment:
Instead of just having(reentrant) you should check if the shim is in the initialization step. For example (in_init || reentrant). Right now, without that check any calls meant to libc made during the initialization will get stuck in the while loop in init() waiting for the initialization to finish.

Reply:
Sure!